content/docs/reference-react-component.md

```diff
@@ -29,111 +29,112 @@ class Welcome extends React.Component {
 }
 ```
 
-The only method you *must* define in a `React.Component` subclass is called [`render()`](#render). All the other methods described on this page are optional.
+המתודה היחידה *שחייב* להגדיר בתת-מחלקה מסוג `React.Component` נקראת [`render()`](#render). כל שאר המתודות המתוארות בדף זה הן אופציונליות.
 
-**We strongly recommend against creating your own base component classes.** In React components, [code reuse is primarily achieved through composition rather than inheritance](/docs/composition-vs-inheritance.html).
+**אנחנו ממליצים בחום נגד יצירת קומפוננטות-בסיס משלכם.** בקומפוננטות React, [שימוש חוזר בקוד מושג בעיקר על-ידי קומפוזיציה מאשר על-ידי הורשה] (/docs/composition-vs-inheritance.html).
 
->Note:
+>הערה:
 >
->React doesn't force you to use the ES6 class syntax. If you prefer to avoid it, you may use the `create-react-class` module or a similar custom abstraction instead. Take a look at [Using React without ES6](/docs/react-without-es6.html) to learn more.
+>React לא כופה על שימוש בתחביר מחלקות של גרסת ES6. אם אתם מעדיפים להימנע ממנו, אתם רשאיים להשתמש במודול `create-react-class` או הרחבה מותאמת דומה, במקום. העיפו מבט ב-[שימוש ב-React ללא גרסת ES6](/docs/react-without-es6.html) כדי ללמוד עוד.
 
-### The Component Lifecycle {#the-component-lifecycle}
+### קומפוננטת מעגל-החיים {#the-component-lifecycle}
 
-Each component has several "lifecycle methods" that you can override to run code at particular times in the process. **You can use [this lifecycle diagram](http://projects.wojtekmaj.pl/react-lifecycle-methods-diagram/) as a cheat sheet.** In the list below, commonly used lifecycle methods are marked as **bold**. The rest of them exist for relatively rare use cases.
+לכל קומפוננטה קיימות מספר "מתודות של מעגל חיים" שניתן לדרוס כדי להריץ קוד בזמנים מסוימים בתהליך. **אתם יכולים להשתמש [בדיאגרמת מעגל-החיים הזו](http://projects.wojtekmaj.pl/react-lifecycle-methods-diagram/) כשליף.** ברשימה מטה, מתודות מעגל-חיים שבשימוש נפוץ יותר, מסומנות **במודגש**. שאר המתודות קיימות עבור שימושים נדירים יותר.
 
-#### Mounting {#mounting}
+#### עיגון (Mounting) {#mounting}
 
-These methods are called in the following order when an instance of a component is being created and inserted into the DOM:
+קריאה למתודות הבאות מתבצעות בסדר הזה כאשר מופע של קומפוננטה נוצר ומוכנס לתוך ה-DOM:
 
 - [**`constructor()`**](#constructor)
 - [`static getDerivedStateFromProps()`](#static-getderivedstatefromprops)
 - [**`render()`**](#render)
 - [**`componentDidMount()`**](#componentdidmount)
 
->Note:
+>הערה:
 >
->These methods are considered legacy and you should [avoid them](/blog/2018/03/27/update-on-async-rendering.html) in new code:
->
->- [`UNSAFE_componentWillMount()`](#unsafe_componentwillmount)
+>המתודות האלו נחשבות למורשת וכדאי [להימנע מהן](/blog/2018/03/27/update-on-async-rendering.html) בקוד חדש:
```

Comment:
```suggestion
>המתודות האלו נחשבות לישנות וכדאי [להימנע מהן](/blog/2018/03/27/update-on-async-rendering.html) בקוד חדש:
```

מורשת is confusing even tho its the translation for legacy, but the meaning is ישנות

Reply:
yeah, it was a bit confusing. I was looking for referneces for this word on other translated pages, some used "מורשת" and some "ישנות". I think perhaps it should be added into the glossary (which is not well extensive TBH)